src/openrct2/ride/ride.c

```diff
@@ -8226,3 +8226,127 @@ LocationXY16 ride_get_rotated_coords(sint16 x, sint16 y, sint16 z)
 
     return rotatedCoords;
 }
+
+void fix_ride_entrance_and_exit_locations()
+{
+    sint32 rideIndex;
+    Ride * ride;
+
+    FOR_ALL_RIDES(rideIndex, ride)
+    {
+        for (sint32 stationIndex = 0; stationIndex < MAX_STATIONS; stationIndex++)
+        {
+            LocationXY8       entranceLoc        = ride->entrances[stationIndex];
+            LocationXY8       exitLoc            = ride->exits[stationIndex];
+            uint8             entranceExitHeight = ride->station_heights[stationIndex];
+            bool              fixEntrance        = false;
+            bool              fixExit            = false;
+            rct_map_element * mapElement;
```

Comment:
we're not modifying `mapElement`, it could be marked as `const`.

Reply:
Ok, but we are ++'ing and assigning this pointer. Is that not a problem?

Reply:
This is pointer-to-const, not const-pointer, so not a problem